src/recipe_dictionary.cpp

```diff
@@ -32,10 +32,121 @@ const recipe &recipe_dictionary::get_uncraft( const itype_id &id )
 std::vector<const recipe *> recipe_subset::search( const std::string &txt ) const
 {
     std::vector<const recipe *> res;
-
+    std::vector<std::string> subqueries;
+    std::vector<std::string> requirement_qualities,
+        names,
+        components,
+        tools,
+        result_qualities,
+        skills;
+    std::string pattern = txt; //Copy the search string.
```

Comment:
Because I iteratively alter the value in order to make the parsing easier. 

I'd have to change the method declaration to make the copy implicitly.